examples/tv-casting-app/tv-casting-common/src/CastingServer.cpp

```diff
@@ -320,24 +321,81 @@ CastingServer::ContentLauncherLaunchURL(TargetEndpointInfo * endpoint, const cha
 
 void CastingServer::DeviceEventCallback(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
 {
+    bool runPostCommissioning           = false;
+    chip::NodeId targetPeerNodeId       = 0;
+    chip::FabricIndex targetFabricIndex = 0;
     if (event->Type == DeviceLayer::DeviceEventType::kBindingsChangedViaCluster)
     {
+        ChipLogProgress(AppServer, "CastingServer::DeviceEventCallback kBindingsChangedViaCluster received");
         if (CastingServer::GetInstance()->GetActiveTargetVideoPlayer()->IsInitialized())
         {
+            ChipLogProgress(AppServer,
+                            "CastingServer::DeviceEventCallback already connected to video player, reading server clusters");
             CastingServer::GetInstance()->ReadServerClustersForNode(
                 CastingServer::GetInstance()->GetActiveTargetVideoPlayer()->GetNodeId());
         }
+        else if (CastingServer::GetInstance()->udcInProgress)
+        {
+            ChipLogProgress(AppServer,
+                            "CastingServer::DeviceEventCallback UDC is in progress while handling kBindingsChangedViaCluster");
+            CastingServer::GetInstance()->udcInProgress = false;
+            if (CastingServer::GetInstance()->mTargetVideoPlayerNumIPs > 0)
+            {
+                TargetVideoPlayerInfo * connectableVideoPlayerList =
+                    CastingServer::GetInstance()->ReadCachedTargetVideoPlayerInfos();
+                if (connectableVideoPlayerList == nullptr || !connectableVideoPlayerList[0].IsInitialized())
+                {
+                    ChipLogProgress(AppServer, "No cached video players found");
```

Comment:
should we be exiting here? Seems like 2 lines later a connectableVideoPlayerList nullptr would cause a crash

Reply:
Done. Added return statement. Also upgraded this to a ChipLogError.